R/FilterStateLogical.R

```diff
@@ -248,13 +262,12 @@ LogicalFilterState <- R6::R6Class( # nolint
         FALSE
       } else if (private$is_choice_limited) {
         TRUE
-      } else if (!isTRUE(private$get_keep_na()) && private$na_count > 0) {
-        TRUE
       } else if (all(private$choices_counts > 0)) {
         TRUE
-      } else if (private$get_selected() == FALSE && private$choices_counts["FALSE"] == 0L) {
+      } else if (isFALSE(all.equal(private$get_selected(), private$choices)) &&
```

Comment:
We don't want to flag as "filtered" when everything is selected in the different order
```r
all.equal(c(FALSE, TRUE), c(TRUE, FALSE)) 
#> [1] "2 element mismatches"

setequal(c(FALSE, TRUE), c(TRUE, FALSE)) 
#> [1] TRUE
```

```suggestion
      } else if (setequal(private$get_selected(), private$choices) &&
```